### PR TITLE
[libpq] update to 14.4 and pass flags in msbuild

### PR DESCRIPTION
**Pull request by Neumann-A**

### CODE REVIEW

File: ports/libpq/patches/fix-configure.patch [hunk 1]
```diff
@@ -4,10 +4,10 @@ index fba79ee..4fadb94 100644
 +++ b/configure.ac
 @@ -19,7 +19,7 @@ m4_pattern_forbid(^PGAC_)dnl to catch undefined macros
  
- AC_INIT([PostgreSQL], [14.1], [pgsql-bugs@lists.postgresql.org], [], [https://www.postgresql.org/])
+ AC_INIT([PostgreSQL], [14.4], [pgsql-bugs@lists.postgresql.org], [], [https://www.postgresql.org/])
  
 -m4_if(m4_defn([m4_PACKAGE_VERSION]), [2.69], [], [m4_fatal([Autoconf version 2.69 is required.
-+m4_if(m4_defn([m4_PACKAGE_VERSION]), [2.69], [], [m4_warn([unsupported],[Autoconf version 2.69 is required.
++m4_if(m4_defn([m4_PACKAGE_VERSION]), [2.69], [], [m4_warn([Autoconf version 2.69 is required.
```
Comment (Neumann-A) [inline ports/libpq/patches/fix-configure.patch:10]:
```suggestion
+m4_if(m4_defn([m4_PACKAGE_VERSION]), [2.69], [], [m4_warn([unsupported],[Autoconf version 2.69 is required.
```

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
Feature: 
client、icu、tcl test successfully in the x64-linux triplet

bonjour、python test successfully in the x64-osx triplet

nls、xml、xslt、lz4、openssl、zlib test successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

File: ports/libpq/patches/windows/macro-def.patch [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+index cac7570ea..d0ca1243c 100644
+--- a/src/include/common/checksum_helper.h
++++ b/src/include/common/checksum_helper.h
+@@ -18,6 +18,11 @@
+ #include "common/sha2.h"
+ #include "port/pg_crc32c.h"
+ 
++#if defined(CHECKSUM_TYPE_NONE)
++// winioctl.h defines CHECKSUM_TYPE_NONE to 0 as a macro. 
++#undef CHECKSUM_TYPE_NONE
```
Comment (ras0219-msft) [inline ports/libpq/patches/windows/macro-def.patch:11]: I'm concerned about doing this in a public header file -- could we instead do:
```cpp
#pragma push_macro("CHECKSUM_TYPE_NONE")
#ifdef CHECKSUM_TYPE_NONE
#undef CHECKSUM_TYPE_NONE
#endif
CHECKSUM_TYPE_NONE,
#pragma pop_macro("CHECKSUM_TYPE_NONE")
```
in the enum definition?

**Review by ras0219-msft — DISMISSED** [review]
This LGTM other than the macro concern. Thanks!

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**